skipper.go

```diff
@@ -827,6 +832,7 @@ func run(o Options, sig chan os.Signal, idleConnsCH chan struct{}) error {
 	log.Infof("Expose metrics in %s format", metricsKind)
 	mtrOpts := metrics.Options{
 		Format:                             metricsKind,
+		MetricsBackend:                     o.MetricsBackend,
```

Comment:
(see my other comment for the proposed solution in metrics.NewMetrics, too)

I think we should rather check in this scope whether o.MetricsBackend (or simply o.Metrics is also a possible name) is nil, and if not, then use it, or if yes, only then call metrics.NewMetrics. Sg like:

```
mtr := o.MetricsBackend
if mtr == nil {
    mtrOpts := metrics.Options{/* ... all without the metrics backend */}
    mtr = metrics.NewMetrics(mtrOpts)
}
```

Reply:
I'm gonna review my code but are you sure this is the preferred way? 

For me it's going to be a little weird because is going to take place on the `skipper` package and will be transparent on the `metrics` package itself. Which means other functions that call `NewMetrics` directly (right now it's just `NewDefaultHandler`) could have a totally different behaviour.

Reply:
Of course, cannot be 100% sure, but how I read this change, what happens is that we only create a metrics object from the metrics package, if we don't provide one from on our own. The current state reflects exactly this, while the previous one kind of made this more hidden. This is why I like it better like this, after you updated.